compiler/src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -396,7 +396,7 @@ class Definitions {
     def newArrayMethod(implicit ctx: Context): TermSymbol = DottyArraysModule.requiredMethod("newArray")
 
   // TODO: Remove once we drop support for 2.12 standard library
-  private[this] lazy val isNewCollections = ctx.base.staticRef("scala.collection.IterableOnce".toTypeName).exists
+  private def isNewCollections = ctx.settings.YnewLibrary.value
```

Comment:
Actually maybe we should do:
```scala
def isNewCollections = ctx.settings.YnewLibrary.value || 
  ctx.base.staticRef("scala.collection.IterableOnce".toTypeName).exists
```
The flag is only useful when we are compiling the sources of 2.13 library with the 2.12 library on the classpath (e.g. community build, tests)